src/platform/qpg6100/KeyValueStoreManagerImpl.cpp

```diff
@@ -22,13 +22,84 @@
  */
 
 #include <platform/KeyValueStoreManager.h>
+#include <support/CodeUtils.h>
 
 namespace chip {
 namespace DeviceLayer {
 namespace PersistedStorage {
 
 KeyValueStoreManagerImpl KeyValueStoreManagerImpl::sInstance;
 
+CHIP_ERROR KeyValueStoreManagerImpl::_Get(const char * key, void * value, size_t value_size, size_t * read_bytes_size,
+                                          size_t offset_bytes)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    qvStatus_t result;
+    size_t actual_read_bytes;
+
+    if ((key == NULL) || (value == NULL))
+    {
+        ExitNow(err = CHIP_ERROR_INVALID_ARGUMENT);
+    }
+
+    result = qvCHIP_KvsGet(key, value, value_size, &actual_read_bytes, offset_bytes);
+    if (result == QV_STATUS_BUFFER_TOO_SMALL)
+    {
+        err = CHIP_ERROR_BUFFER_TOO_SMALL;
+    }
+    else if (result != QV_STATUS_NO_ERROR)
+    {
+        err = CHIP_ERROR_PERSISTED_STORAGE_FAILED;
```

Comment:
nit: https://github.com/project-chip/connectedhomeip/blob/master/src/include/platform/KeyValueStoreManager.h#L76 documentation does not list `CHIP_ERROR_PERSISTED_STORAGE_FAILED` error code as valid.